src/align.rs

```diff
@@ -86,3 +90,79 @@ def_align!(8, Align8);
 def_align!(16, Align16);
 def_align!(32, Align32);
 def_align!(64, Align64);
+
+/// A `Vec` that uses a 64 byte aligned allocation.
+///
+/// Only works with `Copy` types so that we don't have to handle drop logic.
+#[derive(Default)]
+pub struct AlignedVec64<T: Copy> {
+    inner: Vec<MaybeUninit<Align64<[u8; 64]>>>,
+    len: usize,
+    _phantom: std::marker::PhantomData<T>,
+}
+
+impl<T: Copy> AlignedVec64<T> {
+    /// Returns the number of elements in the vector.
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    pub fn as_ptr(&self) -> *const T {
+        self.inner.as_ptr().cast()
+    }
+
+    pub fn as_mut_ptr(&mut self) -> *mut T {
+        self.inner.as_mut_ptr().cast()
+    }
+
+    /// Extracts a slice containing the entire vector.
+    pub fn as_slice(&self) -> &[T] {
+        // SAFETY: The first `len` elements of the buffer have been initialized.
+        unsafe { slice::from_raw_parts(self.as_ptr(), self.len) }
```

Comment:
We can't know `T` is transmutable from `u8`s.  There'd have to be some sort of bound, like `T: bytemuck::AnyBitPattern`.